### PR TITLE
backend/fix : added api to update fare policy

### PR DESCRIPTION
**Pull request by VD321**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/CommonAPIs/package.yaml [hunk 1]
```diff
@@ -80,6 +80,9 @@ dependencies:
   - deriving-aeson
   - extra
   - servant-multipart
+  - beam-core
+  - beam-postgres
```
Comment (khuzema786) [inline Backend/app/dashboard/CommonAPIs/package.yaml:84]: Not required.

File: Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Merchant.hs [hunk 2]
```diff
@@ -648,6 +653,74 @@ newtype UpdateFPPerExtraKmRateReq = UpdateFPPerExtraKmRateReq
...
+    baseDistance :: Maybe Meters,
+    baseFare :: Maybe Money,
+    deadKmFare :: Maybe Money,
+    waitingCharge :: Maybe WaitingCharge,
+    waitingChargeInfo :: Maybe WaitingChargeInfo,
+    freeWaitingTime :: Maybe Minutes,
+    nightShiftCharge :: Maybe NightShiftCharge
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
```
Comment (khuzema786) [inline Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Merchant.hs:676]: ToSchema instance not required for this, only FromJSON is required

File: Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Merchant.hs [hunk 3]
```diff
@@ -648,6 +653,74 @@ newtype UpdateFPPerExtraKmRateReq = UpdateFPPerExtraKmRateReq
...
+  hideSecrets = identity
+
+data WaitingChargeInfo = WaitingChargeInfo
+  { freeWaitingTime :: Minutes,
+    waitingCharge :: WaitingCharge
+  }
+  deriving (Generic, Eq, Show, ToJSON, FromJSON, ToSchema, Read)
+
+instance FromField WaitingChargeInfo where
+  fromField = fromFieldJSON
```
Comment (khuzema786) [inline Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Merchant.hs]: All this instance not required.

File: Backend/app/dashboard/CommonAPIs/package.yaml (hunk 1 above)
Comment (VD321) [reply Backend/app/dashboard/CommonAPIs/package.yaml]: Done

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**